### ISSUE TITLE
Media file size is too small for important old documents scans

### ISSUE THREAD

**Issue by Gawrion** [HuubMons/HuMo-genealogy#48]
Hey. It's me again. Today I dug up my ancestor's baptismal and birth certificate from 1901. The certificate is written in cyrillic, in russian, because at that time Poland was under russian rule. I haven't deciphered the entire text yet and the quality of the uloaded document is extremely important for reading. The default limit of 2MB is causing my whole very important document to be very pixelated and not readable.

I recommend raising the limit to 5MB and changing height/width factors - now they are fixed, u should calculate it somehow from size limit to be (proportional) universal to any byte limits. I raised it to 5MB and now my file is ok. Alternatively the program should check the upper limit for media upload (php settings) and allow to set it up to this limit somewhere in settings - but default should not be 2MB (it's to low).

**Comment by HuubMons** [comment]
The reason for a 2MB limit is speed. I just clicked both images, the 5MB version was showed with a delay.
If a page contains multiple images of 5MB, the page will be shown quit slow.

We probably need an option to upload larger pictures.
Work around: just upload this picture directly into the picture folder using FTP. I didn't test it, but I do think this will work.

**Comment by chs-59** [comment]
Hi everyone ,
in the recent developer version pictures are resized to maxwh 3840x2160px, no file size limit. But indeed there should be an option to shut off resizing and/or give some format options. If people use small pictures, resizing is not really necessary. On the other hand there are those who don't expect a loss of quality when they upload a picture. Right now there is no documentation of the resize function, am I right?
The shut off selection can easily go where the thumbnail selection resides in index_admin.php close to the library test. That makes sense because if no graphic library is installed, resizing will be skipped anyway.

**Comment by HuubMons** [comment]
In the past months I allready tried to make some changes.
When I used larger thumbnails, there were problems with photobook: the page will be slow.

There are users that uploaded very large pictures (10 - 20 MB?). For a website it's not needed to add such a huge picture, and some pages were slow because of the picture size. This was the reason the automatically resize was added for original pictures.

Maybe we need multiple sizes:
- Original size
- Medium size thumbnail
- Small size thumbnail

And maybe directory structure must be changed to (folder with original files, and other folders for generated thumbnails).

**Comment by chs-59** [comment]
Hi everyone,

I'm not familiar with the recent code base of HuMo-genealogy any more but as far as I remember the lack of performance within the "photobook" section was a result of the heavy database access that is invoked with **every** action there. The code always reads all pictures from Db, checking all privacy rules and getting all descriptions. With some thousands of pictures this can spent a lot of time even if you just click on the next page. In my [HuMo-gen-Fork](https://github.com/chs-59/HuMo-gen-Server) the scan is done only once per session and the result is stored in a session key. That speeds up the pages a lot. I don't think the size of the thumbnails will make the difference.

**Comment by HuubMons** [comment]
Oh, interesting!

It seems it's a fork of HuMo-genealogy 6.x.
HuMo-genealogy is now 7.x, the database is rebuild. I didn't want to use GEDCOM number as references anymore, so it's rebuild to use proper index numbers.

I don't think it's possible to merge anything from this fork. I have changed too many scripts in the last versions.

**Comment by chs-59** [comment]
No, indeed. I just wanted to add some of my experiences about the thumbnail performance to this discussion.
